src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.exeptions.NonAnnotationException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NonAnnotationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(BetDao.class)) {
+                    field.set(instance, Factory.getBetDao());
+                }
+                if (field.getType().equals(UserDao.class)) {
+                    field.set(instance, Factory.getUserDao());
+                }
```

Comment:
use `if-else if-else` construction and `&&` to refactor this part (add checking if implementation classes have `@Dao`)